extras/rya.geoindexing/src/test/java/org/apache/rya/indexing/geotemporal/GeoTemporalProviderTest.java

```diff
@@ -0,0 +1,222 @@
+            "PREFIX time: <tag:rya-rdf.org,2015:temporal#>" +
+            "SELECT * WHERE { " +
+                "?subj <" + tempPred + "> ?time ."+
+                "?subj <" + GeoConstants.GEO_AS_WKT + "> ?loc . " +
+                "?subj2 <" + tempPred + "> ?time2 ."+
+                "?subj2 <" + GeoConstants.GEO_AS_WKT + "> ?loc2 . " +
+                " FILTER(geos:sfContains(?loc, " + geo + ")) . " +
+                " FILTER(time:equals(?time, " + temp + ")) . " +
+            "}";
+        final QuerySegment<EventQueryNode> node = getQueryNode(query);
+        final List<EventQueryNode> nodes = provider.getExternalSets(node);
+        assertEquals(1, nodes.size());
+    }
+
+    @Test
+    public void twoNode_test() throws Exception {
+        final ValueFactory vf = new ValueFactoryImpl();
+        final Value geo = vf.createLiteral("Point(0 0)", GeoConstants.XMLSCHEMA_OGC_WKT);
+        final Value temp = vf.createLiteral(new TemporalInstantRfc3339(2015, 12, 30, 12, 00, 0).toString());
+        final URI tempPred = vf.createURI(URI_PROPERTY_AT_TIME);
+        final String query =
+            "PREFIX geo: <http://www.opengis.net/ont/geosparql#>" +
+            "PREFIX geos: <http://www.opengis.net/def/function/geosparql/>" +
+            "PREFIX time: <tag:rya-rdf.org,2015:temporal#>" +
+            "SELECT * WHERE { " +
+                "?subj <" + tempPred + "> ?time ."+
+                "?subj <" + GeoConstants.GEO_AS_WKT + "> ?loc . " +
+                "?subj2 <" + tempPred + "> ?time2 ."+
+                "?subj2 <" + GeoConstants.GEO_AS_WKT + "> ?loc2 . " +
+                " FILTER(geos:sfContains(?loc, " + geo + ")) . " +
+                " FILTER(time:equals(?time, " + temp + ")) . " +
+                " FILTER(geos:sfContains(?loc2, " + geo + ")) . " +
+                " FILTER(time:equals(?time2, " + temp + ")) . " +
+            "}";
+        final QuerySegment<EventQueryNode> node = getQueryNode(query);
+        final List<EventQueryNode> nodes = provider.getExternalSets(node);
+        assertEquals(2, nodes.size());
+    }
+
+    @Test
+    public void twoSubjectMultiFilter_test() throws Exception {
+        final ValueFactory vf = new ValueFactoryImpl();
+        final Value geo = vf.createLiteral("Point(0 0)", GeoConstants.XMLSCHEMA_OGC_WKT);
+        final Value temp = vf.createLiteral(new TemporalInstantRfc3339(2015, 12, 30, 12, 00, 0).toString());
+        final URI tempPred = vf.createURI(URI_PROPERTY_AT_TIME);
+        final String query =
+            "PREFIX geo: <http://www.opengis.net/ont/geosparql#>" +
+            "PREFIX geos: <http://www.opengis.net/def/function/geosparql/>" +
+            "PREFIX time: <tag:rya-rdf.org,2015:temporal#>" +
+            "SELECT * WHERE { " +
+                "?subj <" + tempPred + "> ?time ."+
+                "?subj <" + GeoConstants.GEO_AS_WKT + "> ?loc . " +
+                " FILTER(geos:sfContains(?loc, " + geo + ")) . " +
```

Comment:
What happens in this case?  I don't remember.  Does the EventQueryNode just have multiple geo filters and temporal filters?  Seems like you support that.